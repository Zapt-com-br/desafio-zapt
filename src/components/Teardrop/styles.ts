import styled, { css } from 'styled-components';
import media from 'styled-media-query';

import { TeardropProps } from '.';

const wrapperModifiers = {
  small: () => css`
    width: 5.2rem;
    height: 5.2rem;

    ${media.greaterThan('medium')`
      width: 6.2rem;
      height: 6.2rem;
    `}
  `,

  normal: () => css`
    width: 6.8rem;
    height: 6.8rem;

    ${media.greaterThan('medium')`
      width: 10rem;
      height: 10rem;
    `}
  `,
};

export const Wrapper = styled.div<TeardropProps>`
  ${({ theme, color, size }) => css`
    color: ${theme.colors[color!]};

    ${!!size && wrapperModifiers[size]}
  `}
`;
