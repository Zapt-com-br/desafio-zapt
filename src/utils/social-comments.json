{
  "title": "O que est√£o falando nas redes sociais",
  "comments": [
    {
      "id": 1,
      "comment": "N√£o da para parar de comer.. esse final de semana pedi 20 pasteis da Ilma e acabaram TODOS em um s√≥ dia... O pessoal aqui de casa ficou viciado de um n√≠vel que terei que comprar toda semana... Assim n√£o tem academia que secura rsrsrsrs Obrigado Zapt.",
      "name": "ferri-gui",
      "image": "/images/social-comments/ferri-gui.svg"
    },
    {
      "id": 2,
      "comment": "Estava precisando achar algum lugar que vendia produtos que v√™em direto da fazenda... a Zapt salvou.. e ainda paquei um pre√ßo muito mais barato que supermecado üôå",
      "name": "paolaC",
      "image": "/images/social-comments/paola.svg"
    },
    {
      "id": 3,
      "comment": "Todos os vinhos da vin√≠cula Bueno est√£o dispon√≠veis no catalogo da Zapt a um pre√ßo incriv√©l... n√£o percam essa oportunidade...",
      "name": "Galv√£o",
      "image": "/images/social-comments/galvao.svg"
    },
    {
      "id": 4,
      "comment": "Nunca tinha achado os produtos na Lush depois que ela saiu do Brasil.. s√≥ a Zapt para conseguir importalos a pre√ßo mais barato do que vendiam aqui! Muito obrigado, recomendo sempre a todos os amigos.",
      "name": "LiciaR",
      "image": "/images/social-comments/liciar.svg"
    },
    {
      "id": 5,
      "comment": "Ontem chegou uma pizza que comprei na Zapt e me lembrou as que minha av√≥ fazia... sensacional!",
      "name": "shitz",
      "image": "/images/social-comments/shitz.svg"
    },
    {
      "id": 6,
      "comment": "Nossa cada dia tem uma oferta mais legal do que a outra.. Al√©m de produtos de qualidade o pr√©co √© imbat√≠vel",
      "name": "rafael",
      "image": "/images/social-comments/rafael.svg"
    },
    {
      "id": 7,
      "comment": "Achei uma promo√ß√£o incr√≠vel de roupinha de beb√™, foi o que salvou o enxoval do Pedro.",
      "name": "gust",
      "image": "/images/social-comments/gust.svg"
    },
    {
      "id": 8,
      "comment": "Meu dono finalmente conseguiu achar uma comida que eu gosto! Agora vou conseguir engordar e ser feliz.",
      "name": "Cookie",
      "image": "/images/social-comments/cookie.svg"
    },
    {
      "id": 9,
      "comment": "N√£o conhecia a Zapt.. um amigo meu indicou... falou para eu dar uma olhada nos pre√ßos das cervejas artesanais.. incr√≠vel.",
      "name": "duque",
      "image": "/images/social-comments/duque.svg"
    }
  ]
}
