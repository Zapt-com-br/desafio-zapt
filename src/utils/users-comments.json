{
  "title": "O que as pessoas estão achando",
  "comments": [
    {
      "id": 1,
      "image": "/images/user-comments/image-1.png",
      "comment": "Não consigo mais viver sem os produtos do Sítio, a diferença do que eu comprava antes e agora é abissal",
      "user": {
        "avatar": "/images/user-comments/patricia.svg",
        "name": "Patricia Godoy",
        "city": "São Paulo",
        "state": "São Paulo"
      }
    },
    {
      "id": 2,
      "image": "/images/user-comments/image-2.png",
      "comment": "Depois que você começa a usar produtos de qualidade é bem difícil voltar atrás...",
      "user": {
        "avatar": "/images/user-comments/marcinho.svg",
        "name": "Marcinho Ribeiro",
        "city": "Belo Horizonte",
        "state": "Minas Gerais"
      }
    },
    {
      "id": 3,
      "image": "/images/user-comments/image-1.png",
      "comment": "Não consigo mais viver sem os produtos do Sítio, a diferença do que eu comprava antes e agora é abissal",
      "user": {
        "avatar": "/images/user-comments/patricia.svg",
        "name": "Patricia Godoy",
        "city": "São Paulo",
        "state": "São Paulo"
      }
    },
    {
      "id": 4,
      "image": "/images/user-comments/image-2.png",
      "comment": "Depois que você começa a usar produtos de qualidade é bem difícil voltar atrás...",
      "user": {
        "avatar": "/images/user-comments/marcinho.svg",
        "name": "Marcinho Ribeiro",
        "city": "Belo Horizonte",
        "state": "Minas Gerais"
      }
    }
  ]
}
