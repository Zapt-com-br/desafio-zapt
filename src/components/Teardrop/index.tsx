import * as S from './styles';

export type TeardropProps = {
  color?: 'primary' | 'secondary';
  size?: 'small' | 'normal';
};

const Teardrop = ({ color = 'primary', size = 'normal' }: TeardropProps) => (
  <S.Wrapper color={color} size={size}>
    <svg
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 62 62"
      role="img"
      aria-label="Teardrop"
    >
      <path
        d="M0 31C0 13.88 13.88 0 31 0c17.12 0 31 13.88 31 31 0 17.12-13.88 31-31 31H0V31z"
        fill="currentColor"
      />
    </svg>
  </S.Wrapper>
);

export default Teardrop;
