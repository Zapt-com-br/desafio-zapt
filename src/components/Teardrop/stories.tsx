import { Story, Meta } from '@storybook/react/types-6-0';

import Teardrop, { TeardropProps } from '.';

export default {
  title: 'Teardrop',
  component: Teardrop,
} as Meta;

export const Default: Story<TeardropProps> = (args) => <Teardrop {...args} />;
