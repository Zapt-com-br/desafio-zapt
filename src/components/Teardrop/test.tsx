import { render, screen } from 'utils/test-utils';

import Teardrop from '.';

describe('<Teardrop />', () => {
  it('should render with primary color and normal size by default', () => {
    const { container } = render(<Teardrop />);

    expect(screen.getByLabelText(/Teardrop/i).parentElement).toHaveStyle({
      color: '#334bc8',
      width: '6.8rem',
      height: '6.8rem',
    });
    expect(
      screen.getByLabelText(/Teardrop/i).parentElement
    ).toHaveStyleRule('width', '10rem', { media: '(min-width:768px)' });
    expect(
      screen.getByLabelText(/Teardrop/i).parentElement
    ).toHaveStyleRule('height', '10rem', { media: '(min-width:768px)' });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with secondary color when passed', () => {
    render(<Teardrop color="secondary" />);

    expect(screen.getByLabelText(/Teardrop/i).parentElement).toHaveStyle({
      color: '#ff5b58',
    });
  });

  it('should render with small size when passed', () => {
    render(<Teardrop size="small" />);

    expect(screen.getByLabelText(/Teardrop/i).parentElement).toHaveStyle({
      width: '5.2rem',
      height: '5.2rem',
    });
    expect(
      screen.getByLabelText(/Teardrop/i).parentElement
    ).toHaveStyleRule('width', '6.2rem', { media: '(min-width:768px)' });
    expect(
      screen.getByLabelText(/Teardrop/i).parentElement
    ).toHaveStyleRule('height', '6.2rem', { media: '(min-width:768px)' });
  });
});
