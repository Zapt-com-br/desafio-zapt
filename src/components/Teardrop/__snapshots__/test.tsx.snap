// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Teardrop /> should render with primary color and normal size by default 1`] = `
.c0 {
  color: #334bc8;
  width: 6.8rem;
  height: 6.8rem;
}

@media (min-width:768px) {
  .c0 {
    width: 10rem;
    height: 10rem;
  }
}

<div
  class="c0"
  color="primary"
>
  <svg
    aria-label="Teardrop"
    fill="none"
    role="img"
    viewBox="0 0 62 62"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 31C0 13.88 13.88 0 31 0c17.12 0 31 13.88 31 31 0 17.12-13.88 31-31 31H0V31z"
      fill="currentColor"
    />
  </svg>
</div>
`;
